README.md

```diff
@@ -41,7 +42,8 @@ func main() {
     "subscription": "projects/myproject/subscriptions/mysubscription"
   }`)
 
-	e, err := pubsub.UnmarshalMessagePublishedData(data)
```

Comment:
My preference for the custom function was in the context of performing more actions beyond just the base unmarshalling, e.g., handling the base64 decoding. Agreed that a single "task" wrapped in this function is not helpful, but if we might add this in the future it's worth considering what usage pattern we want to encourage.

Reply:
OK.
Need a full feature request for this custom unmarshaling idea. To me, it seems like there will be minimal benefit for this library to add decorator functionality than providing a tighter interface.

base64 decoding will come with https://github.com/googleapis/google-cloudevents-go/issues/49